src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -50,28 +52,30 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    -rational == Rational(-rational.numer, rational.denom)
   }
 
   property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+    abs((left + right).toDouble - (left.toDouble + right.toDouble)) <= KINDA_SMALL_NUMBER

```

Comment:
What the reason to decries precision using doubles.
Why don't to compare the result `numer` with left's + right's numer?
The same for `denom`. 
Thus the test is more clear and you don't need to use magic numbers such as `KINDA_SMALL_NUMBER`.